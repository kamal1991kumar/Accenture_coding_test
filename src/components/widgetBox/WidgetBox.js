import "./WidgetBox.scss";
import React from "react";
import PropTypes from "prop-types";
export const WidgetBox = ({
  title,
  body,
  id,
  handleEditClick,
  widgetIndex,
}) => (
  <div className="widgetBox" data-testid="widgetBox" data-id={id}>
    <hgroup className="widgetBox__hgroup">
      <h3 className="widgetBox__heading" data-testid="title">
        {id}-{title}
      </h3>
      <button
        className="widgetBox__btn"
        data-testid="edit"
        onClick={() => handleEditClick(widgetIndex)}
      >
        <img
          width="15"
          src="https://img.icons8.com/material-outlined/20/999999/edit--v1.png"
          alt="edit"
        />
      </button>
    </hgroup>
    <p className="widgetBox__text" data-testid="body">
      {body}
    </p>
  </div>
);
WidgetBox.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  handleEditClick: PropTypes.func.isRequired,
  widgetIndex: PropTypes.number.isRequired,
};
WidgetBox.defaultProps = {
  widgetIndex: 1,
};
