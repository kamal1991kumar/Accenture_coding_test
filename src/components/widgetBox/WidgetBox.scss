
.widgetBox {
  border: solid 1px #ccc;
  padding: 15px;
  box-shadow: 0 0 5px rgba(0,0,0,.5);
  border-radius: 5px;
  margin: 8px 0;
  &__hgroup {
    display: flex;
    padding-bottom: 5px;
  }
  &__heading {
    font-size: 12px;
    font-weight: normal;
    text-transform: capitalize;
    flex: 1;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
  &__btn {
    border: none;
    background: none;
    cursor: pointer;
  }
  &__text {
    font-size: 10px;
    line-height: 15px;
  }
}