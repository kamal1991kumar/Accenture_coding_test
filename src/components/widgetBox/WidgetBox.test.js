import React from "react";
import { render, fireEvent } from "@testing-library/react";
import WidgetBox from "./index";

const props = {
  title: "Kamal",
  body: "My name is kamal",
  id: 1,
  handleEditClick: jest.fn(),
  widgetIndex: 0,
};

describe("WidgetBox Component", () => {
  test("Should render on screen", () => {
    const { container } = render(<WidgetBox {...props} />);
    expect(container).toMatchSnapshot();
  });

  test("Should match the title", () => {
    const { getByTestId } = render(<WidgetBox {...props} />);
    expect(getByTestId("title").textContent).toBe(`${props.id}-${props.title}`);
  });

  test("Should match the body", () => {
    const { getByTestId, container } = render(<WidgetBox {...props} />);
    expect(container).toMatchSnapshot();
    expect(getByTestId("body").textContent).toBe(props.body);
  });

  test("Should be clickable on edit button", () => {
    const { getByTestId } = render(<WidgetBox {...props} />);
    const btn = getByTestId("edit");
    fireEvent.click(btn);
    expect(props.handleEditClick.mock.calls.length).toEqual(1);
  });
});
