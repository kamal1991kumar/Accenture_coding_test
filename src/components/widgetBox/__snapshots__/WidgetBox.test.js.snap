// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetBox Component Should match the body 1`] = `
<div>
  <div
    class="widgetBox"
    data-id="1"
    data-testid="widgetBox"
  >
    <hgroup
      class="widgetBox__hgroup"
    >
      <h3
        class="widgetBox__heading"
        data-testid="title"
      >
        1
        -
        Kamal
      </h3>
      <button
        class="widgetBox__btn"
        data-testid="edit"
      >
        <img
          alt="edit"
          src="https://img.icons8.com/material-outlined/20/999999/edit--v1.png"
          width="15"
        />
      </button>
    </hgroup>
    <p
      class="widgetBox__text"
      data-testid="body"
    >
      My name is kamal
    </p>
  </div>
</div>
`;

exports[`WidgetBox Component Should render on screen 1`] = `
<div>
  <div
    class="widgetBox"
    data-id="1"
    data-testid="widgetBox"
  >
    <hgroup
      class="widgetBox__hgroup"
    >
      <h3
        class="widgetBox__heading"
        data-testid="title"
      >
        1
        -
        Kamal
      </h3>
      <button
        class="widgetBox__btn"
        data-testid="edit"
      >
        <img
          alt="edit"
          src="https://img.icons8.com/material-outlined/20/999999/edit--v1.png"
          width="15"
        />
      </button>
    </hgroup>
    <p
      class="widgetBox__text"
      data-testid="body"
    >
      My name is kamal
    </p>
  </div>
</div>
`;
